models/Element/AbstractElement.php

```diff
@@ -474,15 +488,49 @@ public function getVersions()
         return [];
     }
 
+    /**
+     * @return array
+     */
+    protected function getBlockedVars(): array
+    {
+
+        $hasPrefix = in_array(Service::getElementType($this), ['asset', 'document']);
+
+        if($this->isInDumpState()) {
+            $blockedVars = [...$this->blockedVars, $hasPrefix ? 'o_dirtyFields' : 'dirtyFields'];
+            $this->removeInheritedProperties();
+
+            return $blockedVars;
+        }
+
+        return [...$this->blockedVars, $hasPrefix ? 'o_children' : 'children', 'properties'];
+
+    }
+
     /**
      * {@inheritdoc}
      */
     public function __sleep()
     {
-        $parentVars = parent::__sleep();
-        $blockedVars = ['dependencies'];
+        return array_diff(parent::__sleep(), $this->getBlockedVars());
```

Comment:
```suggestion

        if ($this->isInDumpState()) {
            // this is if we want to make a full dump of the object (eg. for a new version), including children for recyclebin
            $this->removeInheritedProperties();
        }

        return array_diff(parent::__sleep(), $this->getBlockedVars());
```

Reply:
I have made changes according to the review